lib/AST/AutoDiff.cpp

```diff
@@ -366,6 +366,13 @@ GenericSignature autodiff::getDifferentiabilityWitnessGenericSignature(
   return derivativeGenSig;
 }
 
+TangentSpace TangentSpace::getTuple(TupleType *tupleTy) {
+  if (tupleTy->getElements().size() == 0) {
```

Comment:
`getNumElements` is more efficient. Or you could do `tupleTy->isVoid()`, which does the check for you.